modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java

```diff
@@ -103,23 +92,53 @@ public static long addListTypeDefinition(
 			listTypeEntry -> listTypeEntries.put(
 				listTypeEntry.getKey(), listTypeEntry));
 
-		for (int i = 0; i < objectStatesJSONArray.length(); i++) {
-			JSONObject objectStateJSONObject =
-				objectStatesJSONArray.getJSONObject(i);
+		if (!ArrayUtil.isEmpty(stateFlowSettings)) {
+			JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+				JSONFactoryUtil.looseSerializeDeep(
+					stateFlowSettings[0].getValue()));
 
-			String key = objectStateJSONObject.getString("key");
+			JSONArray objectStatesJSONArray = jsonObject.getJSONArray(
+				"objectStates");
 
-			if (listTypeEntries.containsKey(key)) {
-				listTypeEntries.remove(key);
+			for (int i = 0; i < objectStatesJSONArray.length(); i++) {
+				JSONObject objectStateJSONObject =
+					objectStatesJSONArray.getJSONObject(i);
 
-				continue;
+				String key = objectStateJSONObject.getString("key");
+
+				if (listTypeEntries.containsKey(key)) {
+					listTypeEntries.remove(key);
+
+					continue;
+				}
+
+				listTypeEntryLocalService.addListTypeEntry(
+					null, userId, listTypeDefinition.getListTypeDefinitionId(),
+					key,
+					Collections.singletonMap(LocaleUtil.getDefault(), key));
 			}
+		}
 
-			listTypeEntryLocalService.addListTypeEntry(
-				null, userId, listTypeDefinition.getListTypeDefinitionId(), key,
-				Collections.singletonMap(LocaleUtil.getDefault(), key));
+		ObjectFieldSetting[] defaultValueSettings = ArrayUtil.filter(
+			objectField.getObjectFieldSettings(),
+			objectFieldSetting -> StringUtil.equals(
+				objectFieldSetting.getName(),
+				ObjectFieldSettingConstants.NAME_DEFAULT_VALUE));
+
+		if (ArrayUtil.isEmpty(defaultValueSettings)) {
+			return listTypeDefinition.getListTypeDefinitionId();
 		}
 
+		String defaultValue = String.valueOf(
+			defaultValueSettings[0].getValue());
+
+		listTypeEntries.remove(defaultValue);
```

Comment:
Keep the pattern of previous remove method.

`if (listTypeEntries.containsKey(key)) {
					listTypeEntries.remove(key);
					continue;
				}`

Reply:
Done!! :white_check_mark: